packages/assets-controllers/src/MultichainBalancesController/MultichainBalancesController.ts

```diff
@@ -0,0 +1,390 @@
+  balances: {
+    [account: string]: {
+      [asset: string]: {
+        amount: string;
+        unit: string;
+      };
+    };
+  };
+};
+
+/**
+ * Default state of the {@link MultichainBalancesController}.
+ */
+export const defaultState: MultichainBalancesControllerState = { balances: {} };
+
+/**
+ * Returns the state of the {@link MultichainBalancesController}.
+ */
+export type MultichainBalancesControllerGetStateAction =
+  ControllerGetStateAction<
+    typeof controllerName,
+    MultichainBalancesControllerState
+  >;
+
+/**
+ * Updates the balances of all supported accounts.
+ */
+export type MultichainBalancesControllerUpdateBalancesAction = {
+  type: `${typeof controllerName}:updateBalances`;
+  handler: MultichainBalancesController['updateBalances'];
+};
+
+/**
+ * Event emitted when the state of the {@link MultichainBalancesController} changes.
+ */
+export type MultichainBalancesControllerStateChange =
+  ControllerStateChangeEvent<
+    typeof controllerName,
+    MultichainBalancesControllerState
+  >;
+
+/**
+ * Actions exposed by the {@link MultichainBalancesController}.
+ */
+export type MultichainBalancesControllerActions =
+  | MultichainBalancesControllerGetStateAction
+  | MultichainBalancesControllerUpdateBalancesAction;
+
+/**
+ * Events emitted by {@link MultichainBalancesController}.
+ */
+export type MultichainBalancesControllerEvents =
+  MultichainBalancesControllerStateChange;
+
+/**
+ * Actions that this controller is allowed to call.
+ */
+export type AllowedActions =
+  | HandleSnapRequest
+  | AccountsControllerListMultichainAccountsAction;
+
+/**
+ * Events that this controller is allowed to subscribe.
+ */
+export type AllowedEvents =
+  | AccountsControllerAccountAddedEvent
+  | AccountsControllerAccountRemovedEvent;
+
+/**
+ * Messenger type for the MultichainBalancesController.
+ */
+export type MultichainBalancesControllerMessenger =
+  RestrictedControllerMessenger<
+    typeof controllerName,
+    MultichainBalancesControllerActions | AllowedActions,
+    MultichainBalancesControllerEvents | AllowedEvents,
+    AllowedActions['type'],
+    AllowedEvents['type']
+  >;
+
+/**
+ * {@link MultichainBalancesController}'s metadata.
+ *
+ * This allows us to choose if fields of the state should be persisted or not
+ * using the `persist` flag; and if they can be sent to Sentry or not, using
+ * the `anonymous` flag.
+ */
+const balancesControllerMetadata = {
+  balances: {
+    persist: true,
+    anonymous: false,
+  },
+};
+
+/**
+ * The MultichainBalancesController is responsible for fetching and caching account
+ * balances.
+ */
+export class MultichainBalancesController extends BaseController<
+  typeof controllerName,
+  MultichainBalancesControllerState,
+  MultichainBalancesControllerMessenger
+> {
+  #tracker: BalancesTracker;
+
+  constructor({
+    messenger,
+    state,
+  }: {
+    messenger: MultichainBalancesControllerMessenger;
+    state: MultichainBalancesControllerState;
+  }) {
```

Comment:
as per our controller guidelines https://github.com/MetaMask/core/blob/main/examples/example-controllers/src/pet-names-controller.ts
```suggestion
  constructor({
    messenger,
    state = {},
  }: {
    messenger: MultichainBalancesControllerMessenger;
    state?: Partial<MultichainBalancesControllerState>;
  }) {
```

Reply:
done in [bcaee6c](https://github.com/MetaMask/core/pull/4965/commits/bcaee6c42b4c196809fe2c59cd611ca155794b28)